media/js/base/core-datalayer.js

```diff
@@ -25,6 +25,55 @@ if (typeof Mozilla == 'undefined') {
         return pageId ? pageId : pathName.replace(/\/[^\/]*/, '');
     };
 
+    /** Returns page type.
+    * @param {String} path - URL path name fallback if page ID does not exist.
+    * @return {String} GTM page type.
+    */
+    Analytics.getPageType = function(path) {
+        var pathName = path ? path : document.location.pathname;
+
+        if (/firefox\/(desktop|android|ios)\/$/.test(pathName)) {
+            return 'product page';
+        } else if (/firefox\/(hello|sync|accounts|private-browsing)\/$/.test(pathName)) {
```

Comment:
This also seems like something we should discuss, as it will likely be out of date before long unless it's manually maintained, which isn't ideal.